pkg/reconcilermanager/controllers/reconciler_base_test.go

```diff
@@ -601,8 +604,21 @@ func manualChangeResourceLimits(unDep *unstructured.Unstructured, isManaulChange
 	if !isManaulChange {
 		return resultUnDep, nil
 	}
-	if err := unstructured.SetNestedMap(resultUnDep.Object, map[string]interface{}{"memroy": "400Mi", "cpu": "400m"}, "spec", "template", "spec", "containers[0]", "resources", "limits"); err != nil {
-		return nil, err
+	o, err := kinds.ToTypedObject(resultUnDep, core.Scheme)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert unstructured to typed object: %v", err)
+	}
+	dep, ok := o.(*appsv1.Deployment)
+	if !ok {
+		return nil, fmt.Errorf("failed to convert the Object to Deployment type")
+	}
+	resources := &dep.Spec.Template.Spec.Containers[0].Resources
+	resources.Limits = corev1.ResourceList{}
+	resources.Limits[corev1.ResourceMemory] = resource.MustParse("400Mi")
+	resources.Limits[corev1.ResourceCPU] = resource.MustParse("400m")
+	resultUnDep, err = kinds.ToUnstructured(dep, core.Scheme)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert Deployment Object to Unstructured: %v", err)
 	}
 	return resultUnDep, nil
```

Comment:
I'd suggest renaming `resultUnDep` to something a bit more intuitive. Maybe `updatedDeployment`?

Reply:
Done